fotoapparat/src/main/java/io/fotoapparat/parameter/selector/SelectorFunction.java

```diff
@@ -1,15 +1,14 @@
 package io.fotoapparat.parameter.selector;
 
-import java.util.Collection;
-
 /**
- * Function which selects one element from many.
+ * Function which selects a value based on the input
  */
-public interface SelectorFunction<T> {
+public interface SelectorFunction<Input, Output> {
 
     /**
-     * @return element from given collection.
+     * @return selected value.
      */
-    T select(Collection<T> items);
+    Output select(Input input);
```

Comment:
I am not a big fan of having to add `Collection` to all our existing selectors. One idea is to add 2 new interfaces `RangeSelectorFunction` and `CollectionsSelectorFunction` based on this one. Would help only to reduce the addition of the generics the whole time. cc @dmitry-zaitsev